Seal Task Group (#6060)
taskcluster/taskcluster: ui/docs/manual/tasks/groups.mdx

---
filename: tasks/groups.mdx
title: Task Groups
order: 50
---

# Task Groups

Task Groups are a way to group tasks together based on their `taskGroupId`.

> When multiple tasks are being created at once, they often have the same `taskGroupId`.
> This is useful in situations such as when tasks are created for a Github

Reviewer: ```suggestion
When multiple tasks are created simultaneously, they often have the same `taskGroupId`.
This is useful in situations where tasks are created for a Github
```

Author: thanks!

> push event with a [decision task](/docs/manual/design/conventions/decision-task).

Reviewer 2: This has a good bit of overlap with https://docs.taskcluster.net/docs/manual/tasks/taskgroupid-schedulerid.  I like the idea of describing a taskgroup as a "thing" ("entity"? "resource"?) here.  In the existing docs, it's really just an id, and now that it can be sealed and there's a queue.getTaskGroup, that isn't really appropriate anymore.

Author: Yeah, I was not sure if adding sealing to that doc would make sense, so created a new page.
I will move information from that one and drop that (or maybe add redirect if it was linked somewhere)

Tasks from the same Task Group might have different `projectIds`.

## Sealing Task Gropus

Sometimes, it may be necessary to block the entire task group from accepting new tasks.
For example, if multiple push events occur on the same branch in Github
and create separate task groups, we can stop previous runs and seal the entire
task group to prevent currently running decision tasks from adding new ones.
This can help to save resources.

To seal a task group, you can use the `queue.sealTaskGroup` method.
Once a task group is sealed, it will no longer accept new tasks and will throw

> an error if new task is being added to it.

Reviewer 3: nice!!!
